lib/diagnostic.js

```diff
@@ -39,17 +50,35 @@ const any = function (/* array, callback */) {
 // the callback returns a truthy value
 // you may use any feature of JavaScript
 // other than `Array.prototype.filter`
-const select = function (/* array, callback */) {
 
+// same as filter method?
+const select = function (array, callback) {
+    let result = [];
+
+    array.forEach(function (e, i, a) {
+      if (callback(e, i, a)) {
+        result.push(array[i]);
+      }
+    });
+    return result;
+
+  };
 };
```

Comment:
Yes, but take a look at line 60. What are you actually pushing to the array? Isn't `array[i]` the same thing as...`e`? Also be careful of your `}`s. Make sure they match.